src/main/java/seedu/address/model/transaction/Transaction.java

```diff
@@ -75,14 +83,15 @@ public boolean equals(Object other) {
         return otherTransaction.getDate().equals(date)
                 && otherTransaction.getAmount().equals(amount)
                 && otherTransaction.getDirection().equals(direction)
+                && otherTransaction.getAccount().equals(account)
                 && otherTransaction.getDescription().equals(description)
                 && otherTransaction.getCategories().equals(categories);
```

Comment:
A little inconsistency here. if you're using the getters then you should use getters throughout.

e.g.
otherTransaction.getAmount().equals(getAmount())

Reply:
True, i've changed the getters to access the members directly.